### ISSUE TITLE
Bug: NavList SubNav active state

### ISSUE THREAD

**Issue by gracepark** [primer/react#3564]
### Description

Hi there 👋  I'm running into an issue where if a NavList SubNav is closed and there are no `aria-current` children that are active, the active state is still on the SubNav level. 

https://github.com/primer/react/assets/22055162/67e25d6f-34c9-44f8-a6ce-39242c874903

Please let me know if I'm misunderstanding, but I would have expected that if all the `aria-current` in the children were `false` there would be no active marker on the parent of those children. Thank you so much!

### Steps to reproduce

1. Create a NavList with multiple parents that have children
2. Set `aria-current` based on the child route being equal to the current page
3. Open and close different parents after clicking on different children 

### Version

v35.26.1

### Browser

Chrome

**Comment by snehaljos** [comment]
Hi, Can I work on this issue?

**Comment by siddharthkp** [comment]
Related + more context: https://github.com/github/primer/issues/2512

**Comment by harshalkaigaonkar** [comment]
Hey @joshblack / @siddharthkp , i would like to take up this issue, if it's open.

**Comment by siddharthkp** [comment]
@gracepark Can we close this as completed by https://github.com/primer/react/pull/3698

**Comment by gracepark** [comment]
Yes we can @siddharthkp! Closing now, thank you.
